libguard/guard_interface.cpp

```diff
@@ -231,127 +237,82 @@ GuardRecords getAll()
- *
- * @return NULL on success
- *         Throw exception on failure
- *
- */
-static void deleteRecord(GuardRecord record, int recordPos, int posOfLastRecord)
-{
-    GuardRecord nullGuard;
-    size_t sizeOfGuardRecord = sizeof(nullGuard);
-
-    memset(&nullGuard, 0xFF, sizeOfGuardRecord);
-
-    uint32_t offset = recordPos * sizeOfGuardRecord;
-
-    GuardFile file(guardFilePath);
-    file.erase(offset + headerSize, sizeOfGuardRecord);
-
-    int i = recordPos + 1;
-    while (posOfLastRecord != i)
-    {
-        uint32_t offset1 = i * sizeOfGuardRecord;
-        file.read(offset1 + headerSize, &record, sizeOfGuardRecord);
-        uint32_t offset2 = (i - 1) * sizeOfGuardRecord;
-        file.write(offset2 + headerSize, &record, sizeOfGuardRecord);
-        file.write(offset1 + headerSize, &nullGuard, sizeOfGuardRecord);
-        i++;
-    }
-}
-
 void clear(const EntityPath& entityPath)
 {
     int pos = 0;
-    int delpos = 0;
-    int lastPos = 0;
     GuardRecord existGuard;
     bool found = false;
+    uint32_t offset = 0;
 
     GuardFile file(guardFilePath);
     for_each_guard(file, pos, existGuard)
     {
-        if (existGuard.targetId == entityPath)
+        if ((existGuard.targetId == entityPath) &&
+            (existGuard.recordId != GUARD_RESOLVED))
         {
-            delpos = pos;
+            offset = pos * sizeof(existGuard);
+            existGuard.recordId = GUARD_RESOLVED;
+            file.write(offset + headerSize, &existGuard, sizeof(existGuard));
             found = true;
+            break;
         }
-        lastPos++;
     }
 
     if (!found)
     {
         guard_log(GUARD_ERROR, "Guard record not found");
         throw std::runtime_error("Guard record not found");
     }
-    deleteRecord(existGuard, delpos, lastPos);
 }
 
 void clear(const uint32_t recordId)
 {
     int pos = 0;
-    int delpos = 0;
-    int lastPos = 0;
     GuardRecord existGuard;
     bool found = false;
+    uint32_t offset = 0;
 
     GuardFile file(guardFilePath);
     for_each_guard(file, pos, existGuard)
     {
-        if (be32toh(existGuard.recordId) == recordId)
+        if (be32toh(existGuard.recordId) == recordId &&
+            (existGuard.recordId != GUARD_RESOLVED))
         {
-            delpos = pos;
+            offset = pos * sizeof(existGuard);
+            existGuard.recordId = GUARD_RESOLVED;
+            file.write(offset + headerSize, &existGuard, sizeof(existGuard));
             found = true;
+            break;
         }
-        lastPos++;
     }
 
     if (!found)
     {
         guard_log(GUARD_ERROR, "Guard record not found");
         throw std::runtime_error("Guard record not found");
     }
-
-    deleteRecord(existGuard, delpos, lastPos);
 }
 
 void clearAll()
 {
-    GuardRecords guardRecords;
-    GuardRecord guard;
+    int pos = 0;
+    GuardRecord existGuard;
+    uint32_t offset = 0;
 
-    memset(&guard, 0, sizeof(guard));
+    memset(&existGuard, 0, sizeof(existGuard));
```

Comment:
This is going to break the existing -r option right?  need to make sure re-order the commits with out impacting the existing functionality.
High level you should be are adding feature on top of existibg one , like , here adding new option to invalidate  instead of delete. and clearall you are planing to keep both options right?

Reply:
Made the changes accordingly. -r usecase will not break.
Created invalidateAll function not changing clearAll.